coffee/session.coffee

```diff
@@ -48,10 +48,12 @@ class palava.Session extends @EventEmitter
   # Reset channel and room
   #
   clearConnection: =>
-    @room?.leave()
-    @room = null
-    @channel?.close()
+    if @channel?.isConnected()
```

Comment:
I'm not sure about it, but shouldn't this check simply be part of room.leave()?

Reply:
@farao One could think of it like this: The room is a higher level concept which requires a functional channel. The session, however, integrates all concepts, so it is its responsibility to add this kind of extra check when it's needed. I think, here the isConnected() check is needed, because both scenarios (still able to send leave or connection is already lost) are typical.